### ISSUE TITLE
Question about learning rate

### ISSUE THREAD

**Issue by YingxuanW** [noahzn/Lite-Mono#160]
Thanks for the wonderful works!! It helps me a lot!

I am trying to using your scripts to train the model without pre-training weights, but I found model(640x192) without pre-training weights were evaluated poorly.

Here was my learning rate setting, it works on  model with pre-training weights both 640x192 and 1024x320.

  --lr 0.0001 5e-6 31 0.0001 1e-5 31

Could you please tell me how you set the learning rate when you train the model(640x192) without pre-training weights.

Looking forward to your reply!

Thanks again.
@noahzn

**Comment by YingxuanW** [comment]
I try to see 5e-4 now. Hope it will be helpful.

**Comment by YingxuanW** [comment]
I try to set the lr at 5e-4 now, wish it will be helpful.

**Comment by noahzn** [comment]
@YingxuanW Hi, thanks for your interest in our work. Our paper states," For models trained from scratch an initial learning rate of 5e−4 with a cosine learning rate schedule [26] is adopted, and the training epoch is set to 35."

**Comment by YingxuanW** [comment]
@noahzn I tried to set the lr at 5e-4, but still got the bad evaluation result at model without pre-training.
<img width="664" alt="image" src="https://github.com/user-attachments/assets/29e39f30-2208-4970-b587-8bfccdaa9df2">
But the models with pre-training weights has the closest result to your paper.
<img width="653" alt="image" src="https://github.com/user-attachments/assets/929f03e6-3618-48e4-be50-ac8f8f8d45dc">

**Comment by noahzn** [comment]
Hi, it's really strange. Even your training using pre-trained weights is not that good. 
what `batchsize` and `drop_rate` do you use?

**Comment by YingxuanW** [comment]
> Hi, it's really strange. Even your training using pre-trained weights is not that good. what `batchsize` and `drop_rate` do you use?

batchsize = 48, drop_rate = 0.2( default setting)

**Comment by noahzn** [comment]
can you try using batchsize=12?

**Comment by YingxuanW** [comment]
> can you try using batchsize=12?

Ok, thanks again!

**Comment by YingxuanW** [comment]
<img width="741" alt="image" src="https://github.com/user-attachments/assets/69d2c476-bfe3-4e1c-9049-34ab2ad0c020">

I still got this bad evaluation result while I had already set the batchsize=12 before.

<img width="315" alt="image" src="https://github.com/user-attachments/assets/95c652b6-8237-4c57-8255-1e9e058ea814">

# PATHS
        self.parser.add_argument("--data_path",
                                 type=str,
                                 help="path to the training data",
                                 default=os.path.join(file_dir, "kitti_data"))
        self.parser.add_argument("--log_dir",
                                 type=str,
                                 help="log directory",
                                 default="/tmp1")

        # TRAINING options
        self.parser.add_argument("--model_name",
                                 type=str,
                                 help="the name of the folder to save the model in",
                                 default="lite-mono")
        self.parser.add_argument("--split",
                                 type=str,
                                 help="which training split to use",
                                 choices=["eigen_zhou", "eigen_full", "odom", "benchmark"],
                                 default="eigen_zhou")
        self.parser.add_argument("--model",
                                 type=str,
                                 help="which model to load",
                                 choices=["lite-mono", "lite-mono-small", "lite-mono-tiny", "lite-mono-8m"],
                                 default="lite-mono")
        self.parser.add_argument("--weight_decay",
                                 type=float,
                                 help="weight decay in AdamW",
                                 default=1e-2)
        self.parser.add_argument("--drop_path",
                                 type=float,
                                 help="drop path rate",
                                 default=0.2)
        self.parser.add_argument("--num_layers",
                                 type=int,
                                 help="number of resnet layers",
                                 default=18,
                                 choices=[18, 34, 50, 101, 152])
        self.parser.add_argument("--dataset",
                                 type=str,
                                 help="dataset to train on",
                                 default="kitti",
                                 choices=["kitti", "kitti_odom", "kitti_depth", "kitti_test"])
        self.parser.add_argument("--png",
                                 help="if set, trains from raw KITTI png files (instead of jpgs)",
                                 action="store_true")
        self.parser.add_argument("--height",
                                 type=int,
                                 help="input image height",
                                 default=192)
        self.parser.add_argument("--width",
                                 type=int,
                                 help="input image width",
                                 default=640)
        self.parser.add_argument("--disparity_smoothness",
                                 type=float,
                                 help="disparity smoothness weight",
                                 default=1e-3)
        self.parser.add_argument("--scales",
                                 nargs="+",
                                 type=int,
                                 help="scales used in the loss",
                                 default=[0, 1, 2])
        self.parser.add_argument("--min_depth",
                                 type=float,
                                 help="minimum depth",
                                 default=0.1)
        self.parser.add_argument("--max_depth",
                                 type=float,
                                 help="maximum depth",
                                 default=100.0)
        self.parser.add_argument("--use_stereo",
                                 help="if set, uses stereo pair for training",
                                 action="store_true")
        self.parser.add_argument("--frame_ids",
                                 nargs="+",
                                 type=int,
                                 help="frames to load",
                                 default=[0, -1, 1])

        self.parser.add_argument("--profile",
                                 type=bool,
                                 help="profile once at the beginning of the training",
                                 default=True)

        # OPTIMIZATION options
        self.parser.add_argument("--batch_size",
                                 type=int,
                                 help="batch size",
                                 default=12)
        self.parser.add_argument("--lr",
                                 nargs="+",
                                 type=float,
                                 help="learning rates of DepthNet and PoseNet. "
                                      "Initial learning rate, "
                                      "minimum learning rate, "
                                      "First cycle step size.",
                                 default=[0.0005, 5e-6, 31, 0.0005, 1e-5, 31])
        self.parser.add_argument("--num_epochs",
                                 type=int,
                                 help="number of epochs",
                                 default=50)
        self.parser.add_argument("--scheduler_step_size",
                                 type=int,
                                 help="step size of the scheduler",
                                 default=15)

        # ABLATION options
        self.parser.add_argument("--v1_multiscale",
                                 help="if set, uses monodepth v1 multiscale",
                                 action="store_true")
        self.parser.add_argument("--avg_reprojection",
                                 help="if set, uses average reprojection loss",
                                 action="store_true")
        self.parser.add_argument("--disable_automasking",
                                 help="if set, doesn't do auto-masking",
                                 action="store_true")
        self.parser.add_argument("--predictive_mask",
                                 help="if set, uses a predictive masking scheme as in Zhou et al",
                                 action="store_true")
        self.parser.add_argument("--no_ssim",
                                 help="if set, disables ssim in the loss",
                                 action="store_true")
        self.parser.add_argument("--mypretrain",
                                 type=str,
                                 help="if set, use my pretrained encoder")
        self.parser.add_argument("--weights_init",
                                 type=str,
                                 help="pretrained or scratch",
                                 default="pretrained",
                                 choices=["pretrained", "scratch"])
        self.parser.add_argument("--pose_model_input",
                                 type=str,
                                 help="how many images the pose network gets",
                                 default="pairs",
                                 choices=["pairs", "all"])
        self.parser.add_argument("--pose_model_type",
                                 type=str,
                                 help="normal or shared",
                                 default="separate_resnet",
                                 choices=["posecnn", "separate_resnet", "shared"])

        # SYSTEM options
        self.parser.add_argument("--no_cuda",
                                 help="if set disables CUDA",
                                 action="store_true")
        self.parser.add_argument("--num_workers",
                                 type=int,
                                 help="number of dataloader workers",
                                 default=12)

        # LOADING options
        self.parser.add_argument("--load_weights_folder",
                                 type=str,
                                 help="name of model to load")
        self.parser.add_argument("--models_to_load",
                                 nargs="+",
                                 type=str,
                                 help="models to load",
                                 default=["encoder", "depth", "pose_encoder", "pose"])

        # LOGGING options
        self.parser.add_argument("--log_frequency",
                                 type=int,
                                 help="number of batches between each tensorboard log",
                                 default=100)
        self.parser.add_argument("--save_frequency",
                                 type=int,
                                 help="number of epochs between each save",
                                 default=2)

        # EVALUATION options
        self.parser.add_argument("--disable_median_scaling",
                                 help="if set disables median scaling in evaluation",
                                 action="store_true")
        self.parser.add_argument("--pred_depth_scale_factor",
                                 help="if set multiplies predictions by this number",
                                 type=float,
                                 default=1)
        self.parser.add_argument("--ext_disp_to_eval",
                                 type=str,
                                 help="optional path to a .npy disparities file to evaluate")
        self.parser.add_argument("--eval_split",
                                 type=str,
                                 default="eigen",
                                 choices=[
                                    "eigen"],
                                 help="which split to run eval on")
        self.parser.add_argument("--save_pred_disps",
                                 help="if set saves predicted disparities",
                                 action="store_true")
        self.parser.add_argument("--no_eval",
                                 help="if set disables evaluation",
                                 action="store_true")
        self.parser.add_argument("--eval_out_dir",
                                 help="if set will output the disparities to this folder",
                                 type=str)
        self.parser.add_argument("--post_process",
                                 help="if set will perform the flipping post processing "
                                      "from the original monodepth paper",
                                 action="store_true")



@noahzn

**Comment by noahzn** [comment]
@YingxuanW  Can you take a screenshot of your tensorboard visualization? And, which PyTorch version are you using?

**Comment by YingxuanW** [comment]
<img width="936" alt="image" src="https://github.com/user-attachments/assets/87b3dd9f-57f9-4ba1-b80a-5a1d26ff7a72">
<img width="949" alt="image" src="https://github.com/user-attachments/assets/7f971070-6968-4794-8ad1-fe0f2b526e7d">
I used torch==2.2.2

**Comment by noahzn** [comment]
Your loss curves seem not so good. 
Can you try PyTorch 1.7.1 or 1.12.1?

**Comment by YingxuanW** [comment]
<img width="651" alt="image" src="https://github.com/user-attachments/assets/0ed2732e-2273-48a5-83f1-107d246c74ed">
I tried Pytorch 1.12.1, it works!!!

**Comment by noahzn** [comment]
Good to know!!
